operator-framework/src/test/java/io/javaoperatorsdk/operator/performance/RelatedEventPerfTest.java

```diff
@@ -0,0 +1,142 @@
+package io.javaoperatorsdk.operator.performance;
+
+import io.fabric8.kubernetes.api.model.KubernetesResourceList;
+import io.fabric8.kubernetes.api.model.Namespace;
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+import io.fabric8.kubernetes.client.dsl.MixedOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.javaoperatorsdk.jenvtest.junit.EnableKubeAPIServer;
+import io.javaoperatorsdk.operator.Operator;
+import io.javaoperatorsdk.operator.junit.LocallyRunOperatorExtension;
+import io.javaoperatorsdk.operator.processing.event.source.AbstractEventSource;
+import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+@EnableKubeAPIServer(apiServerFlags = {"--min-request-timeout", "1"})
+public class RelatedEventPerfTest {
+
+    KubernetesClient adminClient = new KubernetesClientBuilder().build();
+    private MixedOperation<Counter, KubernetesResourceList<Counter>, Resource<Counter>> counterClient;
+
+    String namespace = "josdk-perftest";
+
+    private static final Logger log = LoggerFactory.getLogger(RelatedEventPerfTest.class);
+    private Operator operator;
+    private AtomicInteger eventCounter = new AtomicInteger();
+
+    private CounterEventSource eventSource = new CounterEventSource(namespace);
+
+    @BeforeEach
+    void beforeEach(TestInfo testInfo) {
+        LocallyRunOperatorExtension.applyCrd(Counter.class, adminClient);
```

Comment:
there is the Junit5 extension that can be used I thin even in this case, see any integratation test ( IT suffix tests) for usage